tests/system_tests/cui/test_cui.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python3
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+"""
+System tests for the fab command line utility.
+"""
+
+import sys
+import pytest
+from pathlib import Path
+import fab.cui.__main__
+
+
+class TestArgsHandling:
+    """Test the command line arguments."""
+
+    def test_args_help(self, capsys: pytest.CaptureFixture) -> None:
+        """Test list source of arguments."""
+
+        with pytest.raises(SystemExit) as exc:
+            fab.cui.__main__.main(["--help"])
+        assert exc.value.code == 0
+
+        captured = capsys.readouterr()
+        assert "--file FILE" in captured.out
+
+    def test_sys_help(
+        self, capsys: pytest.CaptureFixture, monkeypatch: pytest.MonkeyPatch
+    ) -> None:
+        """Test sys.argv alternate source of arguments."""
+
+        monkeypatch.setattr(sys, "argv", ["__main__.py", "--help"])
+
+        with pytest.raises(SystemExit) as exc:
+            fab.cui.__main__.main()
+        assert exc.value.code == 0
+
+        captured = capsys.readouterr()
+        assert "--file FILE" in captured.out
+
+
+class TestUserTarget:
+    """Test with a user-supplied target script."""
+
+    def test_missing_file(self, capsys: pytest.CaptureFixture) -> None:
+        """Test with a missing target file."""
+
+        with pytest.raises(SystemExit) as exc:
+            fab.cui.__main__.main(["--file", "nosuch"])
+        assert exc.value.code == 2
+
+        captured = capsys.readouterr()
+        assert "error: fab file does not exist" in captured.err
+
+    def test_invalid_class_name(
+        self, tmp_path: Path, capsys: pytest.CaptureFixture
+    ) -> None:
+        """Test with an invalid build class name."""
+
+        target = tmp_path / "invalid.py"
+
+        target.write_text(
+            """
+from fab.target.base import FabTargetBase
+class FabBuildInvalidName(FabTargetBase):
+  def run(self, args):
+    pass
+            """
+        )
```

Comment:
`textwrap.dedent()` can be your friend in these circumstances. It allows the text to be indented, in line with the source code, but to be printed as you expect.